src/api/erc20/Erc20Api.ts

```diff
@@ -103,23 +121,38 @@ export class Erc20ApiImpl implements Erc20Api {
   }
 
   public async name({ tokenAddress, networkId }: NameParams): Promise<string> {
+    const name = this._getLocalErc20Property(tokenAddress, 'name')
+    if (name) {
+      return name
+    }
+
     const erc20 = this._getERC20AtAddress(networkId, tokenAddress)
 
     return await erc20.methods.name().call()
   }
 
   public async symbol({ tokenAddress, networkId }: SymbolParams): Promise<string> {
+    const symbol = this._getLocalErc20Property(tokenAddress, 'symbol')
+    if (symbol) {
+      return symbol
+    }
+
     const erc20 = this._getERC20AtAddress(networkId, tokenAddress)
 
     return await erc20.methods.symbol().call()
   }
 
   public async decimals({ tokenAddress, networkId }: DecimalsParams): Promise<number> {
+    const decimals = this._getLocalErc20Property(tokenAddress, 'decimals')
+    if (decimals) {
+      return decimals
+    }
+
     const erc20 = this._getERC20AtAddress(networkId, tokenAddress)
 
-    const decimals = await erc20.methods.decimals().call()
+    const decimalsString = await erc20.methods.decimals().call()
 
-    return Number(decimals)
+    return Number(decimalsString)
```

Comment:
Not sure if `undefined` is possible as a return here but this will return `NaN` if so - maybe better with a try catch and no error handling beyond rebubbling the error

Reply:
and because JS is litty, `typeof NaN === 'number'`

Reply:
Don't think `undefined` is possible, it should just break if result isn't a string according to ABI.
But I wonder if it can return `0x`

Reply:
If it does break, or return something which isn't convertable to a number, we silently ignore it and default to 18 decimals instead.